src/components/SearchBar.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import { useInput } from '../utilities/customHooks';
+import { validate } from '../utilities/helperFunctions';
+
+export default function SearchBar({ onNewSearch = (f) => f }) {
+  const [searchProp, resetSearch] = useInput('');
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
While I want to understand where you're coming from, I think it's helpful to have the error message be stateful since it's being rendered to the UI below the search bar input.  

If I understand try and catch, it's more for exceptions, which, in my opinion, isn't the purpose of input validation in this case.  

As I see it, the first purpose is to inform the user that what they entered isn't valid and why, which I think would be helpful to display within the UI. The second is to catch some invalid input before sending it to the next function, but not necessarily [exhaustively](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation).   

I do really appreciate your comment because it did make me think a bit more deeply about form validation, and I included some changes to make the checks more accessible!

Reply:
Given that, I do understand where you're coming from regarding displaying the error message in the UI. It'll be helpful for the user to know why their input was invalid instead of just throwing an exception and handling it internally. Thanks for clarifying this portion!